jdbc-driver/src/main/java/com/google/cloud/dataproc/jdbc/HiveUrlUtils.java

```diff
@@ -0,0 +1,170 @@
+
+        paramsToMap(params, /* startIndex= */ 1);
+
+        // Get otherSessionConfigs
+        Set<String> nonSessionConfs =
+                ImmutableSet.of(
+                        "projectId",
+                        "region",
+                        "clusterPoolLabel",
+                        "clusterName",
+                        "port",
+                        "httpPath",
+                        "transportMode");
+
+        String otherSessionConfs =
+                PARAMS_MAP.entrySet().stream()
+                        .filter(entry -> !nonSessionConfs.contains(entry.getKey()))
+                        .map(Object::toString)
+                        .collect(Collectors.joining(";"));
+        if (!otherSessionConfs.isEmpty()) {
+            paramBuilder.setOtherSessionConfs(otherSessionConfs);
+        }
+
+        // Client shouldn't pass in the session variables that are already set
+        checkUrl(
+                !PARAMS_MAP.containsKey("httpPath"), "Invalid variable.\nPlease do not include httpPath.");
+        checkUrl(
+                !PARAMS_MAP.containsKey("transportMode"),
+                "Invalid variable.\nPlease do not include transportMode.");
+
+        try {
+            int port =
+                    Integer.valueOf(PARAMS_MAP.getOrDefault("port", String.valueOf(HIVE_DEFAULT_PORT)));
+            paramBuilder.setPort(port);
+        } catch (NumberFormatException e) {
+            throw new InvalidURLException(
+                    String.format(
+                            "'port=%s' Please indicate correct port number or remove the field.",
+                            PARAMS_MAP.get("port")));
+        }
+
+        String clusterName = PARAMS_MAP.get("clusterName");
+        // TODO: check if cluster name is valid / cluster master node name -m or -m-0
+        checkUrl(clusterName != null, "'clusterName=%s' Please indicate cluster name.", clusterName);
+        paramBuilder.setHost(String.format("%s-m", clusterName));
+
+        return paramBuilder.build();
+    }
+
+    /**
+     * Helper method to check condition and throw InvalidURLException
+     *
+     * @param condition the condition to meet
+     * @param msgFmt the error message
+     * @param msgParams optional message parameter
+     * @throws InvalidURLException
+     */
+    private static void checkUrl(boolean condition, String msgFmt, String... msgParams)
+            throws InvalidURLException {
+        if (!condition) {
+            throw new InvalidURLException(String.format(msgFmt, (Object[]) msgParams));
+        }
+    }
+
+    /**
+     * Helper method that turns the field=value pair into map.
+     *
+     * @param params parameters given by the client
+     * @param startIndex start of sessionConfs
+     * @throws InvalidURLException
+     */
+    private static void paramsToMap(String[] params, int startIndex) throws InvalidURLException {
+        PARAMS_MAP = new LinkedHashMap<>();
```

Comment:
Sorry about confusion, I meant that we should make `nonSessionConfs` set a `NON_SESSION_CONFS` constant, not `paramsMap`, may you revert `parmsMap` change back and make `nonSessionConfs` a class constant?